quickevent/app/quickevent/plugins/Event/src/services/oresultsclient.cpp

```diff
@@ -143,4 +147,173 @@ void OResultsClient::sendFile(QString name, QString request_path, QString file)
+	request.setRawHeader("pwd", settings().apiKey().toUtf8());
+	request.setHeader( QNetworkRequest::ContentTypeHeader, "application/xml" );
+	QNetworkReply *reply = m_networkManager->post(request, xml.toUtf8());
+
+	connect(reply, &QNetworkReply::finished, reply, [reply]()
+	{
+		if(reply->error()) {
+			qfError() << "OReuslts.eu [competitor change]:" << reply->errorString();
+		}
+		else {
+			QString msg = reply->readAll();
+			if (msg.contains("BADPWD"))
+				qfError() << "OReuslts.eu [competitor change]: Invalid API key";
+			else if (!msg.contains("OK"))
+				qfError() << "OReuslts.eu [competitor change]: Failed to process request";
+		}
+		reply->deleteLater();
+	});
+}
+
+static void append_list(QVariantList &lst, const QVariantList &new_lst)
+{
+	lst.insert(lst.count(), new_lst);
+}
+
+static bool is_csos_reg(QString &reg)
+{
+	const std::regex csos_registration_regex("[A-Z]{3}[0-9]{4}");
+	return reg.length() == 7 && std::regex_match(reg.toStdString(), csos_registration_regex);
+}
+
+static int mop_start(int runner_start_ms) {
+	int stage_id = getPlugin<RunsPlugin>()->selectedStageId();
+	QDateTime event_start = getPlugin<EventPlugin>()->stageStartDateTime(stage_id);
+	QDateTime runner_start = event_start.addMSecs(runner_start_ms);
+	return event_start.date().startOfDay().msecsTo(runner_start) / 100;
+}
+
+static int mop_run_status_code(
+		int time,
+		bool isDisq,
+		bool isDisqByOrganizer,
+		bool isMissPunch,
+		bool isBadCheck,
+		bool isDidNotStart,
+		bool isDidNotFinish,
+		bool isNotCompeting)
+{
+	if (isNotCompeting)
+		return 99; // NP (Not participating)
+	if (isMissPunch)
+		return 3; //MP (Missing punch)
+	if (isDidNotFinish)
+		return 4; // DNF (Did not finish)
+	if (isDidNotStart)
+		return 20; // DNS (Did not start)
+	if (isBadCheck || isDisqByOrganizer || isDisq)
+		return 5; // DQ (Disqualified)
+	if (time)
+		return 1; // OK
+	return 0; // Unknown
+}
+
+
+void OResultsClient::onCompetitorChanged(int competitor_id)
+{
+	if (competitor_id == 0)
+		return;
+
+	int stage_id = getPlugin<RunsPlugin>()->selectedStageId();
+	qf::core::sql::Query q;
+	q.exec("SELECT competitors.registration, "
+		   "competitors.startNumber, "
+		   "competitors.lastName || ' ' || competitors.firstName AS name, "
+		   "classes.name AS class, "
+		   "runs.siId, "
+		   "runs.disqualified, "
+		   "runs.disqualifiedByOrganizer, "
+		   "runs.misPunch, "
+		   "runs.badCheck, "
+		   "runs.notStart, "
+		   "runs.notFinish, "
+		   "runs.notCompeting, "
+		   "runs.startTimeMs, "
+		   "runs.timeMs "
+		   "FROM runs "
+		   "INNER JOIN competitors ON competitors.id = runs.competitorId "
+		   "LEFT JOIN relays ON relays.id = runs.relayId  "
+		   "INNER JOIN classes ON classes.id = competitors.classId OR classes.id = relays.classId  "
+		   "WHERE competitors.id=" QF_IARG(competitor_id) " AND runs.stageId=" QF_IARG(stage_id), qf::core::Exception::Throw);
+	if(q.next()) {
+		QString registration = q.value(0).toString();
```

Comment:
please use `QVariant Query::value(const QString &field_name) const`, position arguments are very hard to check